File: Q1. Concatenation of Array/getConcatenation.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getConcatenation = function (nums) {
    let len = nums.length
    //need to create 2*len array
    let ans = new Array(len + len).fill(0)

    for (let i = 0; i < len; i++) {
        ans[i] = nums[i]
        ans[i + len] = nums[i]
    }

    return ans
};


//Another solution using spread operator
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getConcatenation = function (nums) {
    return [...nums, ...nums]
};

File: Array-I/Q1. Concatenation of Array/getConcatenation.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int sz = nums.size();

        vector<int> ans(sz*2, 0);

        for(int i=0; i<sz; i++){
            ans[i] = nums[i];
            ans[i+sz] = nums[i];
        }

        return ans;
    }
};